### PR TITLE
[Doc] add 2.3.12 rn

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: docs/release_notes/release-2.3.md [hunk 1]
```diff
@@ -1,5 +1,28 @@
 # StarRocks version 2.3
 
+## 2.3.12
+
+Release date: April 24, 2023
+
+### Improvements
+
+Support implicit conversion if the returned value of an expression can be legally converted to a Boolean value. [# 21792](https://github.com/StarRocks/starrocks/pull/21792)
```
Comment (amber-create) [inline docs/release_notes/release-2.3.md:9]: image
-> Supports

Comment (amber-create) [reply docs/release_notes/release-2.3.md:9]: image
“合法”处理为 legally 不合适，legally 一般是指法律法律上的合法。这句可以改为：
xxx can be properly converted to xxx
或者
xxx can be converted to a valid Boolean value （推荐）

File: docs/release_notes/release-2.3.md [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+
+### Improvements
+
+Supports implicit conversion if the returned value of an expression can be converted to a valid Boolean value. [# 21792](https://github.com/StarRocks/starrocks/pull/21792)
+
+### Bug Fixes
+
+The following bugs are fixed:
+
+- If a user's LOAD privilege is granted at the table level, an error message `Access denied; you need (at least one of) the LOAD privilege(s) for this operation`  is returned at transaction rollback in the event of a load job failure. [# 21129](https://github.com/StarRocks/starrocks/issues/21129)
```
Comment (evelynzhaojie) [inline docs/release_notes/release-2.3.md:15]: 权限写法：LOAD_PRIV

File: docs/release_notes/release-2.3.md [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+- After ALTER SYSTEM DROP BACKEND is executed to drop a BE, the replicas of tables whose replication number is set to 2 on that BE cannot be repaired. In this situation, data loads into these tables fail. [# 20681](https://github.com/StarRocks/starrocks/pull/20681)
+- The error message returned when unsupported data types are specified at table creation provides inaccurate error information. [# 20999](https://github.com/StarRocks/starrocks/issues/20999)
```
Comment (evelynzhaojie) [inline docs/release_notes/release-2.3.md:17]: NPE is returned when an unsupported data type is used in CREATE TABLE.

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Pull request merged**